src/commands/gate/evaluate.ts

```diff
@@ -145,7 +162,48 @@ export class GateEvaluateCommand extends Command {
       })
   }
 
-  private handleEvaluationSuccess(evaluationResponse: EvaluationResponse): number {
+  /**
+   * Evaluate gate rules and return a promise that will behave as follows:
+   * - If the request is successful, the promise will be resolved with the response
+   * - If the request is successful but the status is 'wait', the promise will be rejected after the received wait time (wait_time_ms)
+   * - If the request is not successful, the promise will be rejected after `initialRetryMs`, with an exponential factor that depends on the attempt (exponential backoff).
+   * If the promise is rejected, `retryRequest` will handle the retry immediately.
+   */
+  private async evaluateRulesWithWait(
+    api: APIHelper,
+    evaluateRequest: Payload,
+    attempt?: number
+  ): Promise<AxiosResponse<EvaluationResponsePayload>> {
+    const timePassed = new Date().getTime() - evaluateRequest.startTimeMs
```

Comment:
this measures the time between the beginning of `execute` and this line? This is always going to be _very_ little time, right? Or am I missing something? It seems unnecessary complexity if it's going to be so little time.

Reply:
For the first request it's going to be small, but for retries it might be minutes, it depends on the wait time returned from the service (the processing delay).
The main thing here is `remainingWait`, so we timeout the command if a lot of time has passed since the beginning.

Reply:
I see. The way it's implemented, if `remainingWait` is 0 it seems there'll be a single retry more (because we'll reject with `'wait'`), but maybe that'd be handled by `maxRetryTime: this.timeoutInSeconds * 1000,` ? 

I think at some point we'll need to add proper e2e tests because the logic has become quite custom

Reply:
You're right, apparently `maxRetryTime` doesn't do anything useful in this case. 

I also missed a couple of edge cases: like failing the command if `--fail-if-unavailble` is enabled, or if the timeout was 0, it just retried a lot.

Reply:
Added integration tests.